pkg/solana/txm/pendingtx_test.go

```diff
@@ -32,51 +30,818 @@ func TestPendingTxContext(t *testing.T) {
 			<-processCtx.Done()
 			wg.Done()
 		}()
-		return solana.SignatureFromBytes(sig), cancel
+		return sig, cancel
 	}
 
 	// init inflight txs map + store some signatures and cancelFunc
 	txs := newPendingTxContext()
-	ids := map[solana.Signature]uuid.UUID{}
+	ids := map[solana.Signature]string{}
 	n := 5
 	for i := 0; i < n; i++ {
-		sig, cancel := newProcess(i)
-		id, err := txs.New(sig, cancel)
+		sig, cancel := newProcess()
+		msg := pendingTx{id: uuid.NewString()}
+		err := txs.New(msg, sig, cancel)
 		assert.NoError(t, err)
-		ids[sig] = id
+		ids[sig] = msg.id
 	}
 
 	// cannot add signature for non existent ID
-	require.Error(t, txs.Add(uuid.New(), solana.Signature{}))
+	require.Error(t, txs.AddSignature(uuid.New().String(), solana.Signature{}))
 
 	// return list of signatures
 	list := txs.ListAll()
 	assert.Equal(t, n, len(list))
 
 	// stop all sub processes
 	for i := 0; i < len(list); i++ {
-		id := txs.Remove(list[i])
+		id, err := txs.Remove(list[i])
+		assert.NoError(t, err)
 		assert.Equal(t, n-i-1, len(txs.ListAll()))
 		assert.Equal(t, ids[list[i]], id)
 
 		// second remove should not return valid id - already removed
-		assert.Equal(t, uuid.Nil, txs.Remove(list[i]))
+		id, err = txs.Remove(list[i])
+		require.Error(t, err)
+		assert.Equal(t, "", id)
 	}
 	wg.Wait()
 }
 
+func TestPendingTxContext_new(t *testing.T) {
+	t.Parallel()
+	_, cancel := context.WithCancel(tests.Context(t))
+	sig := randomSignature(t)
+	txs := newPendingTxContext()
+
+	// Create new transaction
+	msg := pendingTx{id: uuid.NewString()}
+	err := txs.New(msg, sig, cancel)
+	require.NoError(t, err)
+
+	// Check it exists in signature map
+	id, exists := txs.sigToID[sig]
+	require.True(t, exists)
+	require.Equal(t, msg.id, id)
+
+	// Check it exists in broadcasted map
+	tx, exists := txs.broadcastedTxs[msg.id]
+	require.True(t, exists)
+	require.Len(t, tx.signatures, 1)
+	require.Equal(t, sig, tx.signatures[0])
+
+	// Check status is Broadcasted
+	require.Equal(t, Broadcasted, tx.state)
+
+	// Check it does not exist in confirmed map
+	tx, exists = txs.confirmedTxs[msg.id]
+	require.False(t, exists)
+
+	// Check it does not exist in finalized map
+	tx, exists = txs.finalizedErroredTxs[msg.id]
+	require.False(t, exists)
+}
+
+func TestPendingTxContext_add_signature(t *testing.T) {
+	t.Parallel()
```

Comment:
Should we include checks on:
- Duplicate signature?
- Including a signature after confirmation?